### PR TITLE
Prosody modules optimizations

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

**Review by sawall — APPROVED** [review]

File: ansible/roles/prosody/files/mod_muc_auth_vpaas.lua [hunk 1]
```diff
@@ -120,16 +145,17 @@ local function validate_vpaas_token(session)
         -- first uuid is the customer id
```
Comment (bgrozev) [inline ansible/roles/prosody/files/mod_muc_auth_vpaas.lua:145]: These comments no longer apply

File: ansible/roles/prosody/files/mod_muc_auth_vpaas.lua [hunk 2]
```diff
@@ -120,16 +145,17 @@ local function validate_vpaas_token(session)
...
         -- a customer can change the public key so the last uuid after /
         -- uniquely identifies the key for that customer
-        local _, tenant_uuid_from_kid, _ = kid:match("^(vpaas%-magic%-cookie%-)(.+)(/.+)$");
-        local _, tenant_uuid = tenant:match("^(vpaas%-magic%-cookie%-)(.+)$")
-        if tenant_uuid_from_kid ~= tenant_uuid then
+        if kid_parse_cache_obj.tenant ~= tenant then
```
Comment (bgrozev) [inline ansible/roles/prosody/files/mod_muc_auth_vpaas.lua:148]: I think this flow doesn't always parse `kid_parse_cache_obj.tenant`?

File: ansible/roles/prosody/files/mod_muc_auth_vpaas.lua (hunk 1 above)
Comment (damencho) [reply ansible/roles/prosody/files/mod_muc_auth_vpaas.lua]: I dropped them.

File: ansible/roles/prosody/files/mod_muc_auth_vpaas.lua (hunk 2 above)
Comment (damencho) [reply ansible/roles/prosody/files/mod_muc_auth_vpaas.lua:142]: I dropped the vpaas_tenants_cache and also fix that kid_parse_cache_obj.tenant is always parsed.

File: ansible/roles/prosody/files/mod_muc_auth_vpaas.lua [hunk 3]
```diff
@@ -45,6 +49,23 @@ end
 
 timer.add_task(CACHE_EXPIRATION_SECONDS, invalidate_cache)
 
+-- gets or creates an entry in the cache for the specified kid
+-- checks is it vpass and extract the tenant (everything before the /) in the kid
+local function get_kid_kid_parse_cache_obj(kid) then
```
Comment (bgrozev) [inline ansible/roles/prosody/files/mod_muc_auth_vpaas.lua:54]: "kid_kid" in the name on purpose?

File: ansible/roles/prosody/files/mod_muc_auth_vpaas.lua [hunk 4]
```diff
@@ -45,6 +49,23 @@ end
...
+    local kid_parse_cache_obj = kid_parse_cache:get(kid);
+    if not kid_parse_cache_obj then
+        kid_parse_cache_obj = {};
+
+        kid_parse_cache_obj.is_vpaas = starts_with(kid, VPAAS_PREFIX);
+
+        local tenant, _ = kid:match("^(.*)%/(.*)$")
```
Comment (bgrozev) [inline ansible/roles/prosody/files/mod_muc_auth_vpaas.lua:61]: Why define the second group if you don't need it?
Should be the same as `local tenant = kid:match("^(.*)%/.*$")`

**Review by bgrozev — APPROVED** [review]

**Pull request merged**